### PR TITLE
Initial refactor of dynamic fields - cleaned up git commit history

### PR DESCRIPTION
**Pull request by wey-chiang**
## Background

Addressed some of the leftover comments by @kostaspap 

## Changes

Refactored DestinationOverride to Overrides everywhere except at the rule engine python module level.
Added unit tests to `test_main.py` and `test_rule.py` for the rules engine

Added reference to alert delivery output to be aligned with the Alert page

## Testing

- `mage test:ci`
- `mage:deploy` with manual testing

### CODE REVIEW

File: internal/core/alert_delivery/api/outputs.go [hunk 1]
```diff
@@ -28,16 +28,24 @@ import (
...
 	// fetch available panther outputs
 	outputs, err := getOutputs()
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the alert outputID
+	alertOutputs := []*outputModels.AlertOutput{}
+	for _, outputID := range alert.OutputIds {
+		if outputID == "SKIP" {
```
Comment (kostaspap) [inline internal/core/alert_delivery/api/outputs.go:42]: Can we make "SKIP" a constant?

File: internal/log_analysis/rules_engine/src/rule.py [hunk 2]
```diff
@@ -352,27 +352,27 @@ def _get_description(self, event: Mapping, use_default_on_exception: bool = True
...
-    def _get_destination_override(self, event: Mapping, use_default_on_exception: bool = True) -> Optional[List[str]]:
-        if not hasattr(self._module, 'destination_override'):
+    def _get_destinations(self, event: Mapping, use_default_on_exception: bool = True) -> Optional[List[str]]:
+        if not hasattr(self._module, 'destinations'):
             return None
 
         try:
-            command = getattr(self._module, 'destination_override')
-            destination_override = self._run_command(command, event, list())
+            command = getattr(self._module, 'destinations')
```
Comment (kostaspap) [inline internal/log_analysis/rules_engine/src/rule.py:360]: This change is backwards incompatible. 
Is the old code included in 1.14? How are we going to deal with this backward incompatibility? Or are we "hiding" this function in 1.14? 

I had put a similar comment in the old PR: https://github.com/panther-labs/panther/pull/2165#discussion_r535878711

File: internal/log_analysis/rules_engine/src/rule.py [hunk 3]
```diff
@@ -520,3 +520,15 @@ def _run_command(self, function: Callable, event: Mapping, expected_type: Any) -
...
+def _rule_id_to_path(rule_id: str) -> str:
+    """Method returns the file path where the rule will be stored"""
+    safe_id = ''.join(x if _allowed_char(x) else '_' for x in rule_id)
+    path = os.path.join(_RULE_FOLDER, safe_id + '.py')
+    return path
+
+
+def _allowed_char(char: str) -> bool:
+    """Return true if the character is part of a valid rule ID."""
+    return char.isalnum() or char in {' ', '-', '.'}
```
Comment (kostaspap) [inline internal/log_analysis/rules_engine/src/rule.py:534]: Hmmm, why do we need that? We have this one: https://github.com/panther-labs/panther/blob/master/internal/log_analysis/rules_engine/src/util.py#L22-L31

**Review by kostaspap — APPROVED** [review]
LGTM

My only concern is the backwards incompatibility issue.

File: internal/log_analysis/rules_engine/src/rule.py (hunk 2 above)
Comment (wey-chiang) [reply internal/log_analysis/rules_engine/src/rule.py:360]: We could attempt to use both `destination_override` and `destinations` and check against both -- however I am not too concerned about this as this was only silently rolled out in v1.14 and not many people are likely using this feature yet.

File: internal/log_analysis/rules_engine/src/rule.py (hunk 3 above)
Comment (wey-chiang) [reply internal/log_analysis/rules_engine/src/rule.py:534]: I think this was left over code from merging, will remove

File: internal/log_analysis/rules_engine/src/rule.py (hunk 2 above)
Comment (wey-chiang) [reply internal/log_analysis/rules_engine/src/rule.py:360]: Reverted this back in favor of consistency and maintaining backwards compatability

**Pull request merged**